packages/contracts/src/contracts/current/test/TestAssetProxyOwner/TestAssetProxyOwner.sol

```diff
@@ -0,0 +1,47 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.24;
+
+import "../../protocol/AssetProxyOwner/AssetProxyOwner.sol";
+
+contract TestAssetProxyOwner is
+    AssetProxyOwner
+{
+    constructor(
+        address[] memory _owners,
+        address[] memory _assetProxyContracts,
+        uint256 _required,
+        uint256 _secondsTimeLocked
+    )
+        public
+        AssetProxyOwner(_owners, _assetProxyContracts, _required, _secondsTimeLocked)
+    {
+    }
+    
+    /// @dev Compares first 4 bytes of byte array to removeAuthorizedAddress function selector.
+    /// @param data Transaction data.
+    /// @return Successful if data is a call to removeAuthorizedAddress.
+    function isFunctionRemoveAuthorizedAddress(bytes memory data)
```

Comment:
Hmm, I think I prefer not having testing code in the production contracts. It's also a bit of a quirky function, the name and return value suggests it returns a bool, but it actually reverts. And its a public function, so it will show up in the ABI.

We could make it internal and expose it with a public wrapper in the testing contract.

But IMHO, in this case the extra layer of indirection for what is essentially a call to readBytes4 and a comparison hurts the code quality more than what we gain from more consistent testing.

Perhaps there is another way we can test this comparison? This would test the whole modifier:

```
function testStoreTransaction(Transaction tx) returns (uint256 id) {...}

function testValidRemoveAuthorizedAddressTx(uint256 id)
   validRemoveAuthorizedAddressTx(id)
   { /* do nothing, revert through modifier */ }
```

Reply:
I like testing modifier this way.